lib/main.dart

```diff
@@ -1,29 +1,32 @@
-import 'package:firebase_core/firebase_core.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:moneyapp/router/go_router.dart';
 import 'common/cubit/transactions_cubit.dart';
 import 'common/source/api_source.dart';
 import 'package:dio/dio.dart';
+import 'common/cubit/login_cubit.dart';
+import 'common/cubit/signup_cubit.dart';
+import 'common/firebase/firebase_module.dart';
 
 Future<void> main() async {
-  WidgetsFlutterBinding.ensureInitialized();
-
-  await Firebase.initializeApp(
-    options: const FirebaseOptions(
-      apiKey: "AIzaSyAzv3b5Txc0jPkTTBXXq9FqUE5YNXt7p8c",
-      appId: "1:4090609795:android:694b1a6f24d9720aec1bb6",
-      messagingSenderId: "4090609795",
-      projectId: "moneyapp-6d288",
-    ),
-  );
+  await setupFirebase(); // Call your firebase setup
 
   final dio = Dio();
   final apiSource = ApiDataSource(dio);
 
   runApp(
-    BlocProvider(
-      create: (context) => TransactionsCubit(apiSource),
+    MultiBlocProvider(
+      providers: [
+        BlocProvider(
+          create: (context) => TransactionsCubit(apiSource),
+        ),
+        BlocProvider(
+          create: (context) => LoginCubit(),
+        ),
+        BlocProvider(
+          create: (context) => SignupCubit(),
```

Comment:
You will not need global MultiBlocProvider. These cubits should be @injectable, and injected in the screen that uses them.